packages/library-base/src/commonMain/kotlin/io/realm/kotlin/internal/RealmListInternal.kt

```diff
@@ -179,6 +185,35 @@ internal class ManagedRealmList<E>(
     override fun delete() = RealmInterop.realm_list_remove_all(nativePointer)
 }
 
+internal fun <E : BaseRealmObject> ManagedRealmList<E>.query(query: String, args: Array<out Any?>): RealmQuery<E> {
+    val operator: BaseRealmObjectListOperator<E> = operator as BaseRealmObjectListOperator<E>
+    return ObjectQuery.tryCatchCoreException {
+        val queryPointer = inputScope {
+            val queryArgs = convertToQueryArgs(args)
+            RealmInterop.realm_query_parse_for_list(
+                this@query.nativePointer,
+                query,
+                queryArgs
+            )
+        }
+        ObjectBoundQuery(
+            parent,
+            ObjectQuery(
+                operator.realmReference,
+                operator.classKey,
+                operator.clazz,
+                operator.mediator,
+                queryPointer,
+            )
+        )
+//    } catch (exception: Throwable) {
```

Comment:
Whats the use of this comment?

Reply:
I discovered that we had the same logic in the `tryCatchCoreException` so used that and forgot to remove it after verifying all the tests 😅